osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/MonitoringService.java

```diff
@@ -51,6 +55,7 @@ public MonitoringService(GetPeriodicMeterReadsCommandExecutor getPeriodicMeterRe
         this.getPeriodicMeterReadsGasCommandExecutor = getPeriodicMeterReadsGasCommandExecutor;
         this.actualMeterReadsCommandExecutor = actualMeterReadsCommandExecutor;
         this.actualMeterReadsGasCommandExecutor = actualMeterReadsGasCommandExecutor;
+        this.getActualPowerQualityCommandExecutor = getActualPowerQualityCommandExecutor;
```

Comment:
@bvdzwet using autowired instead of constructor injection to avoid this Sonar warning is questionable. Constructor injection has many advantages and is actually preferred by Spring. Also this Sonar issue should be up for debate. How bad is this really? 
The idea here is to refactor the class based on SRP or separation-of-concerns.

Reply:
You may find it questionable. It is common practice in the code base, and it works just fine.
I know constructor injection can have advantages, but in this particular case they seem theoretical and the quantity of constructor arguments looks to be the bigger issue.
Single responsibility principle and separation of concerns: I don't see those in the context of what is discussed here. There are dependencies (the same with constructor or field injection) and the responsibility for injection lies with the Spring set up, either way.

Reply:
Not concluding thid discussion at this moment. Saving it for later to solve it in a broader scope